pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -162,31 +159,33 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 		if clusterBuildStrategy != nil {
 			generatedTaskRun, err = generateTaskRun(build, buildRun, serviceAccount.Name, clusterBuildStrategy.Spec.BuildSteps)
 			if err != nil {
-				reqLogger.Error(err, "Failed to generate TaskRun", "BuildRun", buildRun.Name)
+				err = r.updateBuildRunErrorStatus(buildRun, err)
 				return reconcile.Result{}, err
 			}
 		}
 	} else {
-		log.Error(err, "Unsupported BuildStrategy Kind", "BuildStrategyKind", build.Spec.StrategyRef.Kind)
+		err:= errors.New("unknown strategy " + string(*build.Spec.StrategyRef.Kind))
```

Comment:
from `err:= errors.New...)` to `err := errors.New...)`

Reply:
i think is better if u use `fmt.Errorf("unknown strategy %s", string(*build.Spec.StrategyRef.Kind))`

Reply:
I like it and changed, thanks!